### ISSUE TITLE
IMU preintegration divergence on S3E dataset

### ISSUE THREAD

**Issue by drancon** [zhongshp/Co-LRIO#6]
Hello. Thank you for sharing your great work and dataset.

I've tried running Co-LRIO using ROS2 Humble in Ubuntu 22.04.

The problem is, as soon as I run "run.launch.py" and play one of S3Ev2 bags,
the robots just drift off to the infinity.

I've played the bag files with '--clock' option and the ros parameter yaml file sets 'use_sim_time' to 'True'.

I can't figure out what the problem is.

If you have seen similar issues, could you please give me any advice?

**Comment by drancon** [comment]
I guess the problem is not due to IMU preintegration. It seems to lie in lidar odometry.

**Comment by Mechazo11** [comment]
Hi @drancon you mentioned you ran CoLIRO in ROS 2 Humble, did you make changes to the installation steps? The authors states that CoLIRO only ran in ROS 2 Foxy in Ubuntu 22.04

**Comment by drancon** [comment]
Thanks for the reply, @Mechazo11 
[GTSAM 4.2a8](https://github.com/borglab/gtsam/releases) conflicts with default pcl in ubuntu 22.04.
I modified some parts of CoLRIO so that it can use the recent GTSAM.

And the issue I reported does not occur when I run the package in other computers with ROS 2 Humble and Ubuntu 22.04.
Therefore, I guess the problem came out due to conflict between some packages in my main PC.

In conclusion, if you change some parts of the code to adapt it to the recent GTSAM, you can run CoLRIO in ROS 2 Humble and Ubuntu 22.04.

**Comment by Mechazo11** [comment]
Hi @drancon is your repo publicly available?

**Comment by drancon** [comment]
Hi @Mechazo11, please check https://github.com/ai-mobility-lab/Co-LRIO.
